packages/cli/src/gemini.tsx

```diff
@@ -498,6 +498,20 @@ export async function main() {
 
     // Handle --list-sessions flag
     if (config.getListSessions()) {
+      // Attempt auth for summary generation (gracefully skips if not configured)
```

Comment:
can we move this into generateSummary? That way we only do it if we actually need to auth. would also avoid this implementation detail about getListSessions appearing here.

Reply:
passed optional auth to `generateSummary` 👍